File: ts/example/src/tests/query_tests.ts
// NOTE: This example does not use any Test Frameworks because Vanilla-Mock can work on any Test Frameworks.

import assert from "assert";
import {ExampleQueryStub} from "./query_doubles";
import {ActualQueryUser} from "../query_user";


export function testUsingQueryStub() {
  const query = new ExampleQueryStub("Hello, World!");

  const sut = new ActualQueryUser(query);

  const actual = sut.doSomething();
  assert.strictEqual(actual, "Hello, World!");
}

File: swift/example/Tests/VanillaMockExampleTests/QueryTests.swift
import XCTest
@testable import VanillaMockExample



class QueryTests: XCTestCase {
    func testUsingQueryStub() {
        let query = ExampleQueryStub(willReturn: "Hello, World!")

        let sut = ActualQueryUser(query: query)

        let actual = sut.doSomething()

        XCTAssertEqual(actual, "Hello, World!")
    }
}